iothub/service/src/Http2/HttpMessageHelper2.cs

```diff
@@ -48,7 +50,17 @@ internal static async Task ValidateHttpResponseStatus(HttpStatusCode expectedHtt
         {
             if (expectedHttpStatusCode != responseMessage.StatusCode)
             {
-                throw await ExceptionHandlingHelper.GetDefaultErrorMapping()[responseMessage.StatusCode].Invoke(responseMessage);
+                IReadOnlyDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> defaultErrorMapping =
+                    ExceptionHandlingHelper.GetDefaultErrorMapping();
+                if (defaultErrorMapping.TryGetValue(responseMessage.StatusCode, out Func<HttpResponseMessage, Task<Exception>> mappedException))
+                {
+                    throw await mappedException.Invoke(responseMessage);
+                }
+
+                // Default case for when the mapping of this error code to an exception does not exist yet
```

Comment:
This was orthogonal to the digital twin client work, but I noticed we didn't throw an appropriate exception in the default case